### PR TITLE
feat: impl DROP TABLE on memory catalog based standalone mode

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- Add rule to parse `DROP TABLE` clause
- Implement drop table interface on frontend SQL, datanode gRPC and datanode SQL
- Update integration test correspondingly. 

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
Part of #497

### CODE REVIEW

File: src/servers/src/mysql/writer.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ impl<'a, W: AsyncWrite + Unpin> MysqlResultWriter<'a, W> {
         error: Error,
         w: QueryResultWriter<'a, W>,
     ) -> Result<()> {
-        error!(error; "Failed to execute query '{}'", query);
+        error!("Failed to execute query '{}', error:{:?}", query, error);
```
Comment (killme2008) [inline src/servers/src/mysql/writer.rs:152]: Why change this?

File: src/sql/src/parser.rs [hunk 2]
```diff
@@ -532,4 +566,19 @@ mod tests {
...
+        let result = ParserContext::create_with_dialect(sql, &GenericDialect {});
+        let mut stmts = result.unwrap();
+        assert_eq!(
+            stmts.pop().unwrap(),
+            Statement::DropTable(DropTable {
+                catalog_name: DEFAULT_CATALOG_NAME.to_string(),
+                schema_name: DEFAULT_SCHEMA_NAME.to_string(),
+                table_name: "foo".to_string()
+            })
+        )
```
Comment (killme2008) [inline src/sql/src/parser.rs:582]: Adds a case to test `drop table [schema].[table]`

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/servers/src/mysql/writer.rs (hunk 1 above)
Comment (waynexia) [reply src/servers/src/mysql/writer.rs]: Nice catch! This is added for debugging. I forget to revert it.

**Pull request merged**